tracer/test/Datadog.Trace.Tests/TracerTests.cs

```diff
@@ -524,5 +525,82 @@ public async Task ForceFlush()
 
             agent.Verify(a => a.FlushTracesAsync(), Times.Once);
         }
+
+        [Fact]
+        public void SetUserOnRootSpanDirectly()
```

Comment:
You probably also want to add a test that you set the Tag correctly when the `ISpan` passed in is _not_ a `Span` (e.g. in the version mismatch scenario)

Reply:
I'm unsure of the best way to do this. I could pass a Moq ISpan and that would prove that `SetUser` works with something that is ISpan but not `Span`, but it doesn't seem much of a test. Is there a better way to simulate the version mismatch situation.

Reply:
>Is there a better way to simulate the version mismatch situation

We have some explicit version mismatch integration tests - maybe you could add it to them?

Otherwise, I think creating a `class SpanStub: ISpan` is fine, even though it's not much of a test. It will highlight if we break the contract some time later either way.